nucypher/policy/conditions/context.py

```diff
@@ -12,18 +13,30 @@
 )
 
 USER_ADDRESS_CONTEXT = ":userAddress"
+USER_ADDRESS_EIP712_CONTEXT = ":userAddressEIP712"
+USER_ADDRESS_EIP4361_CONTEXT = ":userAddressEIP4361"
 
 CONTEXT_PREFIX = ":"
 CONTEXT_REGEX = re.compile(":[a-zA-Z_][a-zA-Z0-9_]*")
 
+USER_ADDRESS_SCHEMES = {
+    USER_ADDRESS_CONTEXT: Auth.AuthScheme.EIP712.value,  # default to EIP712
+    USER_ADDRESS_EIP712_CONTEXT: Auth.AuthScheme.EIP712.value,
+    USER_ADDRESS_EIP4361_CONTEXT: Auth.AuthScheme.EIP4361.value,
+}
```

Comment:
I wonder if we could do this substitution on the client side only? I guess not, since it wouldn't be backwards-compatible on the server-side (with existing lingos).

Reply:
Correct. We can't because we have to be backwards compatible with any condition that already uses `:userAddress`.

Reply:
If we want it, we can allow *any* auth scheme for `:userAddress`, i.e. either EIP4361 or EIP712 could be allowed.

Reply:
✔️  - https://github.com/nucypher/nucypher/pull/3508/commits/0214471462e03529d236ec009a2a333f63112ca9